File: it/php/defaultvalues/tests/DefaultValuesTest.php
<?php
namespace Integration\Test;

require __DIR__ . '/../vendor/autoload.php';

use DateTime;

use Integration\Test\Client\ApiClient;
use Integration\Test\Client\Models\WeatherForecast;
use Integration\Test\Client\Models\WeatherForecast_enumValue;

use Microsoft\Kiota\Abstractions\Authentication\AnonymousAuthenticationProvider;
use Microsoft\Kiota\Http\GuzzleRequestAdapter;

use PHPUnit\Framework\TestCase;
class DefaultValuesTest extends TestCase {
    
    public function testDefaultValues() {
        
        $authProvider = new AnonymousAuthenticationProvider();
        $requestAdapter = new GuzzleRequestAdapter($authProvider);
        $requestAdapter->setBaseUrl('http://127.0.0.1:1080');
        $client = new ApiClient($requestAdapter);

        //Call a sample endpoint - not really needed here.
        $serviceResponse = $client->api()->v1()->weatherForecast()->get()->wait();
        $this->assertNotNull($serviceResponse);
        $this->assertEquals(1, sizeof($serviceResponse));

        //Now the real test: create a model class and verify that all properties have the default values.
        $model = new WeatherForecast();
        
        $this->assertEquals(true, $model->getBoolValue());

        $this->assertNotNull($model->getDateOnlyValue());
        $this->assertEquals('1900-01-01', strval($model->getDateOnlyValue()));

        $this->assertNotNull($model->getDateValue());
        //Format the value in ISO8601 (the original value in the model is RFC3339, which has a colon in the timezone offset - this difference is not relevant for the test):
        $this->assertEquals('1900-01-01T00:00:00+0000', $model->getDateValue()->format(DateTime::ISO8601));

        $this->assertNotNull($model->getDateValueLocalTime());
        //Local time in API description is converted to DateTime in local time zone, which depends on the current system. So check only the date/time parts of the value.
        $this->assertStringStartsWith('1900-01-01T00:00:00', $model->getDateValueLocalTime()->format(DateTime::ISO8601));

        $this->assertEquals(25.5, $model->getDecimalValue());
        $this->assertEquals(25.5, $model->getDoubleValue());
        $this->assertEquals(new WeatherForecast_enumValue('one'), $model->getEnumValue());
        $this->assertEquals(25.5, $model->getFloatValue());

        $this->assertNotNull($model->getGuidValue());
        $this->assertEquals('00000000-0000-0000-0000-000000000000', strval($model->getGuidValue()));

        $this->assertEquals(255, $model->getLongValue());
        $this->assertEquals('Test', $model->getSummary());
        $this->assertEquals(15, $model->getTemperatureC());

        $this->assertNotNull($model->getTimeValue());
        $this->assertEquals('00:00:00', strval($model->getTimeValue()));
    }
}
?>

File: it/go/defaultvalues/defaultvalues_test.go
package integrationtest

import (
	"context"
	"integrationtest/client"
	"integrationtest/client/models"
	"testing"
	"time"

	"github.com/google/uuid"
	"github.com/microsoft/kiota-abstractions-go/authentication"
	"github.com/microsoft/kiota-abstractions-go/serialization"
	http "github.com/microsoft/kiota-http-go"
)

func TestMockServerBasic(t *testing.T) {
	httpAdapter, _ := http.NewNetHttpRequestAdapter(&authentication.AnonymousAuthenticationProvider{})
	httpAdapter.SetBaseUrl("http://localhost:1080")
	client := client.NewApiClient(httpAdapter)

	//Call a sample endpoint - not really needed here.
	result, _ := client.Api().V1().WeatherForecast().Get(context.Background(), nil)
	if result == nil {
		t.Errorf("Service response is null\n")
	}
	if len(result) != 1 {
		t.Errorf("Service response does not contain one item\n")
	}

	//Now the real test: create a model class and verify that all properties have the default values.
	model := models.NewWeatherForecast()

	if *(model.GetBoolValue()) != true {
		t.Errorf("boolValue is not true\n")
	}

	if model.GetDateOnlyValue() == nil {
		t.Errorf("dateOnlyValue is null\n")
	}
	if serialization.DateOnly.String(*(model.GetDateOnlyValue())) != "1900-01-01" {
		t.Errorf("dateOnlyValue is not 1900-01-01 but %v\n", serialization.DateOnly.String(*(model.GetDateOnlyValue())))
	}

	if model.GetDateValueLocalTime() == nil {
		t.Errorf("dateValueLocalTime is null\n")
	}
	//Format the datetime in the local format:
	if (*(model.GetDateValueLocalTime())).Format("2006-01-02T15:04:05") != "1900-01-01T00:00:00" {
		t.Errorf("dateValueLocalTime is not 1900-01-01T00:00:00 but %v\n", (*(model.GetDateValueLocalTime())).Format("2006-01-02T15:04:05"))
	}
	if (*(model.GetDateValueLocalTime())).Location() != time.Now().Location() {
		t.Errorf("dateValueLocalTime does not have location %v but %v\n", (*(model.GetDateValueLocalTime())).Location(), time.Now().Location())
	}

	if *(model.GetDecimalValue()) != 25.5 {
		t.Errorf("decimalValue is not 25.5 but %v\n", *(model.GetDecimalValue()))
	}
	if *(model.GetDoubleValue()) != 25.5 {
		t.Errorf("doubleValue is not 25.5 but %v\n", *(model.GetDoubleValue()))
	}
	if *(model.GetEnumValue()) != models.ONE_WEATHERFORECAST_ENUMVALUE {
		t.Errorf("doubleValue is not %v but %v\n", models.ONE_WEATHERFORECAST_ENUMVALUE, *(model.GetEnumValue()))
	}
	if *(model.GetFloatValue()) != 25.5 {
		t.Errorf("floatValue is not 25.5 but %v\n", *(model.GetFloatValue()))
	}

	if model.GetGuidValue() == nil {
		t.Errorf("guidValue is null\n")
	}
	if uuid.UUID.String(*(model.GetGuidValue())) != "00000000-0000-0000-0000-000000000000" {
		t.Errorf("guidValue is not 00000000-0000-0000-0000-000000000000 but %v\n", uuid.UUID.String(*(model.GetGuidValue())))
	}

	if *(model.GetLongValue()) != 255 {
		t.Errorf("longValue is not 255 but %v\n", *(model.GetLongValue()))
	}
	if *(model.GetSummary()) != "Test" {
		t.Errorf("summary is not 'Test' but %v\n", *(model.GetSummary()))
	}
	if *(model.GetTemperatureC()) != 15 {
		t.Errorf("temperatureC is not 15 but %v\n", *(model.GetTemperatureC()))
	}

	if model.GetTimeValue() == nil {
		t.Errorf("timeValue is null\n")
	}
	if serialization.TimeOnly.String(*(model.GetTimeValue())) != "00:00:00" {
		t.Errorf("timeValue is not '00:00:00' but %v\n", serialization.TimeOnly.String(*(model.GetTimeValue())))
	}
}
